cpp/src/parquet/column_reader.cc

```diff
@@ -211,10 +211,15 @@ EncodedStatistics ExtractStatsFromHeader(const H& header) {
     return page_statistics;
   }
   const format::Statistics& stats = header.statistics;
-  if (stats.__isset.max) {
+  // Use the new V2 min-max statistics over the former one if it is filled
```

Comment:
Previously, `page_statistics` will handle min-max separately. This patch changes it to once have all min-max, otherwise, cannot use min-max

Reply:
Yes, page stats without either min or max is corrupted and cannot be used. Check parquet-mr for detail: https://github.com/apache/parquet-mr/blob/5290bd5e0ee5dc30db0576e2bfc6eea335c465cf/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java#L797

Reply:
Although in parquet.thrift, min-max can exist only one. But I think handling it like this is ok

Reply:
So we revert back to previous mode that only has min or max is ok?